src/type_checker/checker.rs

```diff
@@ -623,6 +623,17 @@ impl<B: Backend> TypeChecker<B> {
         args: &[Expr],
         span: Span,
     ) -> Result<Option<TyKind>> {
+        // check if a function names is in use already by another variable
+        match typed_fn_env.get_type_info(&fn_sig.name.value) {
+            Some(_) => {
+                return Err(self.error(
+                    ErrorKind::FunctionNameInUsebyVariable(fn_sig.name.value),
+                    fn_sig.name.span,
+                ))
+            }
+            None => (),
+        };
+
```

Comment:
imo we should do it the other way around, when we create a new variable (var assigment statement) we should check if a function with the same name doesn't already exist.

Reply:
I think functions are being declared statically less than variables, **a function might contain many variables** but not vice versa, checking names on functions' declaration reduces our performance penalty for such extra checks.

Reply:
I guess that makes sense, in this case we're not checking that you CAN'T name a variable name after a function, but rather that if you do you don't end up calling the function with the same name too. It's a bit restrictive but I guess it would be confusing to write such code so I'm in for the compiler deny

Reply:
that makes sense!